tools/depends/configure.ac

```diff
@@ -718,6 +718,156 @@ AC_SUBST(app_rendersystem)
 AC_SUBST(app_winsystem)
 AC_SUBST(ffmpeg_options)
 
+
+# Generate meson cross file
+[
+# Parse and handle CC
+read -r -a CC_ARRAY <<< "$CCACHE $CC"
+if [ ${#CC_ARRAY[@]} -eq 1 ]
```

Comment:
You cannot use bashisms in a POSIX sh script (which is what configure.ac generates).

And here you are simultaneously using:
- bash arrays
- POSIX sh `[` (rather than bash-specific `[[`)
- bash herestring `<<<`
- pre-POSIX `[ "x$foo" = "xbar" ]` tricks for not bombing out when $foo evaluates to something that begins with a dash, and the `[` program is badly written and tries to check a 3-value `strval operator strval` usage using the two-value `-option strval` form

Reply:
feel free to give any info to make it more "correct". i only did enough to get it to function. worked on macos, worked on our linux CI hosts, so thats all i can test really

Reply:
It will work if your system happens to symlink `/bin/sh` to `bash`, which is not at all guaranteed.
It's definitely the case on macOS, but different linuxes do this differently, and some even make it a configuration choice.

Anyway, this PR moves some logic from both target/Makefile and from target/meson-cross-setup.sh, but maybe you could simply run the latter from configure.ac instead of inlining it? The standalone script appears to work fine as it calls out to python in order to do fiddly string splitting and array-izing, and therefore does not care what shell it runs in.

You could also document that anyone who gets really strange syntax errors trying to run `./configure` should mess with root-owned files in `/bin`, I suppose. :)

Reply:
the issue with the script, is it requires python. in a tools depend build, python is build as part of native tools, ie after configure, chicken and egg issue.

Can you provide me with details of how to repro a failure. Im ok to try and make things work, but i need to be able to repro such a setup so i can test and see how it fails.

Reply:
What's your development system? On Debian/Ubuntu, you could install the `dash` package. macOS Catalina has it preinstalled, apparently (and people have theorized as a result that Apple may someday stop using bash for /bin/sh by switching it to dash). In either case, you could run `dash configure` to manually override the interpreter you run the configure script with.

Reply:
yeah i run macos, but can VM whatever i need. I'll try the dash thing and see how it goes i guess.

Reply:
well dash doesnt even work with current master on macos, so im not going to bother fixing something that doesnt work anyway. i'll try an ubuntu vm.

Can i also just confirm, tools/depends really is only intended to build apple or android platforms, not linux. what triggered you looking at this?

Reply:
ok, how about this @eli-schwartz 

https://github.com/fuzzard/xbmc/commit/9d3960a638c379331c63946f59c1c24620b43911

Does that work for your situation?